dipy/io/utils.py

```diff
@@ -223,15 +225,42 @@ def is_reference_info_valid(affine, dimensions, voxel_sizes, voxel_order):
     return all_valid
 
 
+def split_name_with_gz(filename):
+    """
+    Returns the clean basename and extension of a file.
+    Means that this correctly manages the ".nii.gz" extensions.
+
+    Parameters
+    ----------
+    filename: str
+        The filename to clean
+
+    Returns
+    -------
+        base, ext : tuple(str, str)
+        Clean basename and the full extension
+    """
+    base, ext = os.path.splitext(filename)
+
```

Comment:
Can you add here `ext = ext.lower()`.   

Some software do weird stuff (capitalize) and do `myfile.NII.GZ`. your code will failed here because of capitalization.

Reply:
I'm going to fix it in trx-python library too, tomorrow and I'll free!